Source/AliveLibAO/Abe.cpp

```diff
@@ -2258,6 +2260,228 @@ __int16 Abe::TryMountElum_42E600()
+                        FP_FromInteger(0),
+                        field_BC_sprite_scale,
+                        50);
+                }
+
+                if (shootKind != ShootKind::eEverythingElse_0)
+                {
+                    if (shootKind == ShootKind::eHanging_1)
+                    {
+                        field_FC_current_motion = eAbeStates::State_92_ForceDownFromHoist_4297C0;
+                        field_108_bMotionChanged = 1;
+                        field_106_shot = 0;
+                        field_114_gnFrame = 0;
+                    }
+                    else if (shootKind == ShootKind::eRolling_2)
+                    {
+                        if (field_10_anim.field_4_flags.Get(AnimFlags::eBit5_FlipX) == pBullet->field_20 > FP_FromInteger(0))
+                        {
+                            field_FE_next_state = eAbeStates::State_73_RollingKnockback_4291D0;
+                        }
+                        else
+                        {
+                            field_FE_next_state = eAbeStates::State_129_RollingKnockForward_4294F0;
+                        }
+                    }
+                }
+                else
+                {
+                    if (field_1A4_resources.res[0])
+                    {
+                        ToKnockback_422D90(1, 1);
+                    }
+                    else
+                    {
+                        ElumKnockForward_42E780(1);
+                    }
+                    if (field_10_anim.field_4_flags.Get(AnimFlags::eBit5_FlipX) != pBullet->field_20 > FP_FromInteger(0))
+                    {
+                        field_FC_current_motion = eAbeStates::State_128_KnockForward_429330;
+                    }
+                    field_108_bMotionChanged = 1;
+                    field_106_shot = 0;
+
+                    field_B4_velx = field_BC_sprite_scale * FP_FromDouble(7.8);
+                    if (pBullet->field_20 < FP_FromInteger(0))
+                    {
+                        field_B4_velx = -field_BC_sprite_scale * FP_FromDouble(7.8);
+                    }
+                }
+                break;
+            }
+            case BulletType::Type_2:
+            {
+                if (field_BC_sprite_scale == FP_FromDouble(0.5))
+                {
+                    field_106_shot = 0;
+                    field_100_health = FP_FromInteger(1);
+                    return;
+                }
+
+                Path_TLV *pPath = nullptr;
+                while (true)
+                {
+                    pPath = gMap_507BA8.TLV_Get_At_446060(
+                        pPath,
+                        FP_FromInteger(rect.x),
+                        FP_FromInteger(rect.y),
+                        FP_FromInteger(rect.x),
+                        FP_FromInteger(rect.y)
+                    );
+                    if (!pPath)
+                        break;
+
+                    if (pPath->field_4_type != 83)
```

Comment:
use the enum type, looks like ZCover and I think this is the inlined check for ZCover that already exists?